compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -756,6 +754,10 @@ void buildLLVMCPUCodegenPassPipeline(OpPassManager &passManager) {
   passManager.nest<ModuleOp>().addNestedPass<func::FuncOp>(
       createLLVMCPUMaterializeEncodingPass());
   passManager.addNestedPass<ModuleOp>(createBufferizeCopyOnlyDispatchesPass());
+  // Lower softmax early to have it go through the exisiting reduction lowering.
```

Comment:
Right, the reason why I put it there is to get the codegen from aggressive fusion out of the box. I can leave it to decompose but I'm guessing I'll get a perf regression without doing more. I'm trying to get it enabled without perf regression. Then ideally someone should pick it up to do something better on the cpu backend.

Reply:
There shouldnt be a perf regression? Why do you think there would be?

Reply:
because I'm guessing the reductions used to be optimized and now I'm replacing those by a softmax that won't be optimized. I'm guessing based on what is done on GPU backends but I could be wrong. I'm reverting that part if you think it won't be a problem.

Reply:
reverted it

Reply:
I had to move it back otherwise this causes stack explosion